src/dashboard/infrastructure/analytics-4/analytics-4-parameters.php

```diff
@@ -0,0 +1,138 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong -- Needed in the folder structure.
+namespace Yoast\WP\SEO\Dashboard\Infrastructure\Analytics_4;
+
+use Yoast\WP\SEO\Dashboard\Domain\Data_Provider\Parameters;
+
+/**
+ * Domain object to add Analytics 4 specific data to the parameters.
+ */
+class Analytics_4_Parameters extends Parameters {
+
+	/**
+	 * The dimensions to query.
+	 *
+	 * @var array<array<string,string>> $dimensions
+	 */
+	private $dimensions;
+
+	/**
+	 * The dimensions filters.
+	 *
+	 * @var array<string, array<string>> $dimension_filters
+	 */
+	private $dimension_filters;
+
+	/**
+	 * The metrics.
+	 *
+	 * @var array<array<string,string>> $metrics
+	 */
+	private $metrics;
+
+	/**
+	 * The order by.
+	 *
+	 * @var array<array<string,array<string,string>>> $order_by
+	 */
+	private $order_by;
```

Comment:
Most of these can be nullable. But I am not sure that makes sense 🤔  we might want to use this differently in the adapter to build the parameters. So that we don't have to make everything nullable. And if we keep it nullable we need to add that to the @ var things

Reply:
> we might want to use this differently in the adapter to build the parameters. So that we don't have to make everything nullable

Any suggestions? 

Or I can make them have an empty array as default value, so we can still check if they are empty before adding them in the request and not have them nullable.

Reply:
Done the latter, [here](https://github.com/Yoast/wordpress-seo/pull/22023/commits/6b85a6859940feb03f43598613fabaa725fb7fee).